### DISCUSSION TITLE
install/community/kind/online/

### DISCUSSION THREAD

**Discussion by giscus** [DaoCloud/DaoCloud-docs#1858] [Announcements]
# install/community/kind/online/

DaoCloud Enterprise 5.0 下一代容器化平台集大成者，引领云原生浪潮，推动数字化转型。

https://docs.daocloud.io/install/community/kind/online/

**Comment by qiupingdai** [comment]
myIP=$(ip -o route get 1.1.1.1 | cut -d " " -f 7)
./dce5-installer install-app -z -k $myIP:8888
  命令需要添加备注： 当在公有云安装并使用 Node Port 安装时，请直接执行 ./dce5-installer install-app -z -k {公网 IP}:8888

**Comment by liyuerich** [comment]
安装docker 的检查命令

这里建议不用 exit 0 ； 否则直接退出host了。还得我每次重新打开ssh 连接

**Reply by liyuerich** [reply]
set -e
if  [ -x "$(command -v docker )" ] ;then
echo "docker already installed : version = "$(docker -v);
exit 0
fi

建议去掉exit 0

**Comment by Ye-Ting** [comment]
ARM 环境下， Kind 镜像无法使用。建议使用 
`docker.m.daocloud.io/kindest/node:v1.26.6`
修改后的命令为，
`kind create cluster --image docker.m.daocloud.io/kindest/node:v1.26.6 --name=fire-kind-cluster --config=kind_cluster.yaml`

**Comment by samzong** [comment]
```shell
set -e
if [ $(free -g|grep Mem | awk '{print $2}')              -lt 12 ]; then (echo "insufficient memory! (should >=12G)"; exit 1); fi
if [ $(grep 'processor' /proc/cpuinfo |sort |uniq |wc -l) -lt 8 ]; then (echo "insufficient CPU! (should >=8C)"; exit 1); fi
if [ $(df -m / |tail -n 1 | awk '{print $4}') -lt 30720 ]; then (echo "insufficient free disk space of root partition!(should >=30G)"; exit 1); fi
ping daocloud.io -c 1 &> /dev/null || ( echo "no connection to internet! abort." &&  exit 1; )
echo "precheck pass.."
```

可以移除 exit 1 会导致程序退出 ssh 链接.

**Comment by qilongqiu** [comment]
没有arm版的安装器吗

**Reply by qilongqiu** [reply]
macos arm版的

**Comment by yaonq** [comment]
“You can add -j8+ to your installation command to retry from last step” 这是需要安装jdk.8+?

**Reply by yaonq** [reply]
还是步骤8

**Reply by KubeKyrie** [reply]
步骤8

**Comment by paul6668** [comment]
VERSION=v0.22.0 安裝程序出現以下錯誤, 下載不了 istio--linux-amd64.tar.gz 請修復 !!!
1:14AM: + curl --retry 10 --retry-max-time 60 -LO https://files.m.daocloud.io/github.com/istio/istio/releases/download//istio--linux-amd64.tar.gz
1:14AM:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
1:14AM:                                  Dload  Upload   Total   Spent    Left  Speed
1:14AM: 100    19  100    19    0     0     54      0 --:--:-- --:--:-- --:--:--    54
1:14AM: + tar -zxf istio--linux-amd64.tar.gz
1:14AM:
1:14AM: gzip: stdin: not in gzip format
1:14AM: tar: Child returned status 1
1:14AM: tar: Error is not recoverable: exiting now
1:14AM: + cleanup
1:14AM: + echo cleanup
1:14AM: + set +ex
1:14AM: cleanup
1:14AM: [sched-info] Sched shutdown...
1:14AM: --------------------------------------
1:14AM: Unfortunately, the dce5 installation failed at step [8] ....
1:14AM:
1:14AM: You can add -j8+ to your installation command to retry from last step.
1:14AM: --------------------------------------
panic: Failed to Install DCE5, check above error messages. shell execution err is exit status 2

 curl --retry 10 --retry-max-time 60 -LO https://files.m.daocloud.io/github.com/istio/istio/releases/download//istio--linux-amd64.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    19  100    19    0     0     23      0 --:--:-- --:--:-- --:--:--    23

tar -zxf istio--linux-amd64.tar.gz

gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now

**Comment by Mag-FelixFelicis** [comment]
你好 我有一个问题 我不是很理解 这个HAMI是支持一个集群里，NVIDIA GPU设备、海光DCU设备、晟腾NPU设备等同时存在的异构资源池化吗？是把所有存在的上述不同设备池化成一个智算资源池，还是说NVIDIA GPU设备是一个池 然后NPU设备是一个池这样

**Comment by Karl-kid** [comment]
【现象】
- DCE offline-community-v0.43.0-amd64，kind 集群，K8s 1.25.3（kindest-node:v1.25.3），kpanda-controller-manager 镜像 v0.49.0
- kpanda-controller-manager Pod 长期 1/2 CrashLoopBackOff：istio-proxy 正常，manager 容器起不来
- manager 容器启动约 1 秒内 exit 1
- 日志只有一行：
maxprocs: Leaving GOMAXPROCS=8: CPU quota undefined
- termination message 为空，GOTRACEBACK=all 也没有 panic 堆栈
- 离线安装，与网络无关，多次重建均一致
- 其他 kpanda 组件、mcamel 的 MySQL/MinIO/ES/Redis 全部 Running
【排查结果】
- kpanda-webhook-certs Secret 不存在
- kpanda-validating-webhook 的 caBundle 为空
- lease-for-kpanda-webhook-certs 不存在
- RBAC 权限正常：create secrets、patch validatingwebhookconfigurations、leases 全部 yes
- 已排除：
  - Istio sidecar：去掉注入后同样崩
  - webhook 参数：--webhook-enabled 默认就是 true，关不掉，且不是主要变量
  - 环境变量：补 POD_NAMESPACE、WEBHOOK_SERVICE_NAME、WEBHOOK_ORGANIZATION 后同样崩
  - 配置挂载、离线/在线安装差异
【原因】尚未 100% 定位，最怀疑两点：
1. kpanda-controller-manager v0.49.0 与 K8s 1.25.3 版本不兼容
2. DCE v0.43.0 的 kpanda chart 存在已知问题：webhook 证书 Secret 从未生成，manager 启动时初始化 webhook 失败后静默退出（帮助信息里有 waiting to get kpanda webhook secret）
需要社区确认：DCE v0.43.0 对应支持的 Kubernetes 版本和 kindest-node 镜像版本，以及 kpanda v0.49.0 是否有已知的静默 exit 1 问题。
